packages/@tinacms/toolkit/src/components/media/get-mime.js

```diff
@@ -0,0 +1,174 @@
+const VIDEO_FILETYPES = [
+  '3g2',
+  '3gp',
+  'aaf',
+  'asf',
+  'avchd',
+  'avi',
+  'drc',
+  'flv',
+  'm2v',
+  'm3u8',
+  'm4p',
+  'm4v',
+  'mkv',
+  'mng',
+  'mov',
+  'mp2',
+  'mp4',
+  'mpe',
+  'mpeg',
+  'mpg',
+  'mpv',
+  'mxf',
+  'nsv',
+  'ogg',
+  'ogv',
+  'qt',
+  'rm',
+  'rmvb',
+  'roq',
+  'svi',
+  'vob',
+  'webm',
+  'wmv',
+  'yuv',
+]
+
+const IMAGE_FILETYPES = [
+  'ase',
+  'art',
+  'bmp',
+  'blp',
+  'cd5',
+  'cit',
+  'cpt',
+  'cr2',
+  'cut',
+  'dds',
+  'dib',
+  'djvu',
+  'egt',
+  'exif',
+  'gif',
+  'gpl',
+  'grf',
+  'icns',
+  'ico',
+  'iff',
+  'jng',
+  'jpeg',
+  'jpg',
+  'jfif',
+  'jp2',
+  'jps',
+  'lbm',
+  'max',
+  'miff',
+  'mng',
+  'msp',
+  'nef',
+  'nitf',
+  'ota',
+  'pbm',
+  'pc1',
+  'pc2',
+  'pc3',
+  'pcf',
+  'pcx',
+  'pdn',
+  'pgm',
+  'PI1',
+  'PI2',
+  'PI3',
+  'pict',
+  'pct',
+  'pnm',
+  'pns',
+  'ppm',
+  'psb',
+  'psd',
+  'pdd',
+  'psp',
+  'px',
+  'pxm',
+  'pxr',
+  'qfx',
+  'raw',
+  'rle',
+  'sct',
+  'sgi',
+  'rgb',
+  'int',
+  'bw',
+  'tga',
+  'tiff',
+  'tif',
+  'vtf',
+  'xbm',
+  'xcf',
+  'xpm',
+  '3dv',
+  'amf',
+  'ai',
+  'awg',
+  'cgm',
+  'cdr',
+  'cmx',
+  'dxf',
+  'e2d',
+  'egt',
+  'eps',
+  'fs',
+  'gbr',
+  'odg',
+  'svg',
+  'stl',
+  'vrml',
+  'x3d',
+  'sxd',
+  'v2d',
+  'vnd',
+  'wmf',
+  'emf',
+  'art',
+  'xar',
+  'png',
+  'webp',
+  'jxr',
+  'hdp',
+  'wdp',
+  'cur',
+  'ecw',
+  'iff',
+  'lbm',
+  'liff',
+  'nrrd',
+  'pam',
+  'pcx',
+  'pgf',
+  'sgi',
+  'rgb',
+  'rgba',
+  'bw',
+  'int',
+  'inta',
+  'sid',
+  'ras',
+  'sun',
+  'tga',
+  'heic',
+  'heif',
+]
+
+export default function (filename) {
+  const split_arr = filename.split('.')
+  const suffix = split_arr[split_arr.length - 1]
+  if (VIDEO_FILETYPES.includes(suffix)) {
+    return 'video'
+  } else if (IMAGE_FILETYPES.includes(suffix)) {
+    return 'image'
+  } else {
+    return 'file'
+  }
+}
```

Comment:
this was a last resort at determining the file type. was hoping to save type on file `persist` but couldn't figure out how to make the property stick. this just looks at the filename instead and in `media-item.tsx` uses it to determine icon, and then also media-manager.tsx uses it to filter the list of media to only show types relevant to the current filter.